### PR TITLE
[#1540] code cleanup before fix

### PR DESCRIPTION
**Pull request by valllllll2000**
#### Before the PR (what is the issue or what needed to be done)
NA
#### The solution
Before making the fix to repeatable question group bug, make some code cleanup:
* improve readability by extracting some methods
* replace suggested anonymous classes by lambdas 
* some suggested lint fixes 
* extracted a class for repetitions in order to write tests for it
* injecting some classes -> easier to run UI tests as some test results may be flanky due to these dependencies
* rename a method
#### Screenshots (if appropriate)
NA
#### Reviewer Checklist
* [ ] Test plan
* [ ] Copyright header
* [ ] Code formatting
* [ ] Documentation

### CODE REVIEW

**Comment by valllllll2000** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/org/akvo/flow/ui/view/RepeatableGroupIterations.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/akvo-flow-mobile/akvo-flow-mobile/pullRequest?prid=4584694&bid=15397046)

**Comment by valllllll2000** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/org/akvo/flow/ui/view/RepeatableGroupIterations.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/akvo-flow-mobile/akvo-flow-mobile/pullRequest?prid=4584694&bid=15397046)

**Comment by valllllll2000** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/org/akvo/flow/ui/view/RepeatableGroupIterations.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/akvo-flow-mobile/akvo-flow-mobile/pullRequest?prid=4584694&bid=15397046)

File: app/src/main/java/org/akvo/flow/ui/view/RepeatableGroupIterations.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+
+import androidx.annotation.NonNull;
+
+class RepeatableGroupIterations implements Iterable<Integer> {
```
Comment (muloem) [inline app/src/main/java/org/akvo/flow/ui/view/RepeatableGroupIterations.java]: I was not very clear on how this class is going to work but thats better addressed in a call

Comment (valllllll2000) [reply app/src/main/java/org/akvo/flow/ui/view/RepeatableGroupIterations.java]: to be honest I have no idea either, I have extracted this, it was an inner class. Will add some tests to it.

**Pull request merged**